packages/entity-server/src/__tests__/__integration__/fieldsAndFilters.test.ts

```diff
@@ -0,0 +1,255 @@
+        query: { fields: 'fake_field' },
+      });
+
+      const error = JSON.parse(text);
+
+      expect(error).toEqual({
+        error:
+          'Internal server error: Unknown field requested: fake_field, must be one of: id,code,country_code,name,image_url,type,attributes',
+      });
+    });
+
+    it('can fetch an entity with specific fields', async () => {
+      const { text } = await app.get('hierarchy/redblue/KANTO', {
+        query: { fields: 'name,attributes' },
+      });
+
+      const entity = JSON.parse(text);
+      expect(entity).toEqual(getEntityWithFields('KANTO', ['name', 'attributes']));
+    });
+
+    it('can multiple entities with specific fields', async () => {
+      const { text } = await app.post('hierarchy/redblue', {
+        query: { fields: 'code,type,attributes' },
+        body: { entities: ['redblue', 'KANTO', 'VIRIDIAN', 'PKMN_TOWER'] },
+      });
+
+      const entities = JSON.parse(text);
+      expect(entities).toBeArray();
+      expect(sortedByCode(entities)).toEqual(
+        sortedByCode(
+          getEntitiesWithFields(
+            ['redblue', 'KANTO', 'VIRIDIAN', 'PKMN_TOWER'],
+            ['code', 'type', 'attributes'],
+          ),
+        ),
+      );
+    });
+  });
+
+  describe('hierarchy dependant fields', () => {
+    it('parent_code depends on hierarchy used', async () => {
+      const { text: rbText } = await app.get('hierarchy/redblue/BLUE', {
+        query: { fields: 'code,parent_code' },
+      });
+
+      const rbEntity = JSON.parse(rbText);
+      expect(rbEntity).toEqual({ code: 'BLUE', parent_code: 'PALLET' });
+
+      const { text: gsText } = await app.get('hierarchy/goldsilver/BLUE', {
+        query: { fields: 'code,parent_code' },
+      });
+
+      const gsEntity = JSON.parse(gsText);
+      expect(gsEntity).toEqual({ code: 'BLUE', parent_code: 'VIRIDIAN' });
+    });
+
+    it('child_codes depends on hierarchy used', async () => {
+      const { text: rbText } = await app.get('hierarchy/redblue/LAVENDER', {
+        query: { fields: 'code,child_codes' },
+      });
+
+      const rbEntity = JSON.parse(rbText);
+      expect(rbEntity).toEqual({ code: 'LAVENDER', child_codes: ['PKMN_TOWER'] });
+
+      const { text: gsText } = await app.get('hierarchy/goldsilver/LAVENDER', {
+        query: { fields: 'code,child_codes' },
+      });
+
+      const gsEntity = JSON.parse(gsText);
+      expect(gsEntity).toEqual({ code: 'LAVENDER', child_codes: ['LAVENDER_RADIO_TOWER'] });
+    });
+  });
+
+  describe('filter', () => {
+    it('throws error for unknown field', async () => {
+      const { text } = await app.get('hierarchy/redblue/KANTO', {
+        query: { filter: 'fake_field==fake_value' },
+      });
+
+      const error = JSON.parse(text);
+
+      expect(error).toEqual({
+        error:
+          'Internal server error: Unknown filter key: fake_field, must be one of: id,code,country_code,name,image_url,type,attributes_type',
```

Comment:
Here is another error message example - apart from the message format, entity props are prone to adjustment in the future. While it's probably simple to update this test case when that happens, I'd use a bit more generic error message like the following:

```suggestion
          'Unknown filter key: fake_field',
```